data/transactions/verify/txn.go

```diff
@@ -235,12 +244,12 @@ func stxnVerifyCore(s *transactions.SignedTxn, txnIdx int, groupCtx *GroupContex
 			batchVerifier); ok {
 			return nil
 		}
-		return errors.New("multisig validation failed")
+		return signatureErrorf("multisig validation failed")
```

Comment:
@jdtzmn and other reviewers:  The impacted lines are partially out of view, but I'm wondering if there's a historical reason explaining why the error returned by `crypto.MultisigBatchVerify` is _not_ propagated?

* On the surface, it feels like we're needlessly losing error resolution.  
* Tracing git history, it seems to me that the implementation has always ignored the returned error:  https://github.com/algorand/go-algorand/blame/5e4c10832226b9fa974eef38b891a0bafdd00d08/data/transactions/verify/txn.go#L179-L183.

If folks feel the question prompts too much investigation, feel welcomed to tell me so.  Though it feels a bit strange to me, I can live with the logic as is.

Reply:
`LogicSigSanityCheckBatchVerify` performs the same check + also discards the error.  If a change is agreed upon in 1 location, presumably it should be made in both places.

Reply:
`crypto.MultisigBatchVerify` does all the things I dislike in Go code.  It's like the trifecta. Naked returns. Errors defined for single uses. Ignored errors.

Nonetheless, I'm not sure I'd touch it.

Reply:
In fact, we need to be very careful. It's possible that the returned errors are being explicitly tested for somewhere. So, I suppose that we really ought to wrap them in such a way that `Error.Is` and `Error.As` work properly.  I think that means implementing `Unwrap()` on the introduced Signature error type.